Packet++/src/LdapLayer.cpp

```diff
@@ -0,0 +1,210 @@
+		{LdapOperationType::DelRequest,            "DelRequest"},
+		{LdapOperationType::DelResponse,           "DelResponse"},
+		{LdapOperationType::ModifyDNRequest,       "ModifyDNRequest"},
+		{LdapOperationType::ModifyDNResponse,      "ModifyDNResponse"},
+		{LdapOperationType::CompareRequest,        "CompareRequest"},
+		{LdapOperationType::CompareResponse,       "CompareResponse"},
+		{LdapOperationType::AbandonRequest,        "AbandonRequest"},
+		{LdapOperationType::SearchResultReference, "SearchResultReference"},
+		{LdapOperationType::ExtendedRequest,       "ExtendedRequest"},
+		{LdapOperationType::ExtendedResponse,      "ExtendedResponse"},
+		{LdapOperationType::IntermediateResponse,  "IntermediateResponse"},
+		{LdapOperationType::Unknown,               "Unknown"}
+	};
+
+	const std::unordered_map<uint8_t, LdapOperationType> IntToLdapOperationType{
+		{static_cast<uint8_t>(LdapOperationType::BindRequest), LdapOperationType::BindRequest},
+		{static_cast<uint8_t>(LdapOperationType::BindResponse), LdapOperationType::BindResponse},
+		{static_cast<uint8_t>(LdapOperationType::UnbindRequest), LdapOperationType::UnbindRequest},
+		{static_cast<uint8_t>(LdapOperationType::SearchRequest), LdapOperationType::SearchRequest},
+		{static_cast<uint8_t>(LdapOperationType::SearchResultEntry), LdapOperationType::SearchResultEntry},
+		{static_cast<uint8_t>(LdapOperationType::SearchResultDone), LdapOperationType::SearchResultDone},
+		{static_cast<uint8_t>(LdapOperationType::ModifyResponse), LdapOperationType::ModifyResponse},
+		{static_cast<uint8_t>(LdapOperationType::AddRequest), LdapOperationType::AddRequest},
+		{static_cast<uint8_t>(LdapOperationType::AddResponse), LdapOperationType::AddResponse},
+		{static_cast<uint8_t>(LdapOperationType::DelRequest), LdapOperationType::DelRequest},
+		{static_cast<uint8_t>(LdapOperationType::DelResponse), LdapOperationType::DelResponse},
+		{static_cast<uint8_t>(LdapOperationType::ModifyDNRequest), LdapOperationType::ModifyDNRequest},
+		{static_cast<uint8_t>(LdapOperationType::ModifyDNResponse), LdapOperationType::ModifyDNResponse},
+		{static_cast<uint8_t>(LdapOperationType::CompareRequest), LdapOperationType::CompareRequest},
+		{static_cast<uint8_t>(LdapOperationType::CompareResponse), LdapOperationType::CompareResponse},
+		{static_cast<uint8_t>(LdapOperationType::AbandonRequest), LdapOperationType::AbandonRequest},
+		{static_cast<uint8_t>(LdapOperationType::SearchResultReference), LdapOperationType::SearchResultReference},
+		{static_cast<uint8_t>(LdapOperationType::ExtendedRequest), LdapOperationType::ExtendedRequest},
+		{static_cast<uint8_t>(LdapOperationType::ExtendedResponse), LdapOperationType::ExtendedResponse},
+		{static_cast<uint8_t>(LdapOperationType::IntermediateResponse), LdapOperationType::IntermediateResponse}
+	};
+
+	std::string LdapOperationType::toString() const
+	{
+		return LdapOperationTypeToString.at(m_Value);
+	}
+
+	LdapOperationType LdapOperationType::fromIntValue(uint8_t value)
```

Comment:
should call `fromUintValue`? 

also `IntToLdapOperationType` should be `UintToLdapOperationType`?

Reply:
Updated in this commit: https://github.com/seladb/PcapPlusPlus/pull/1412/commits/ff0bcc1a6f6fc74d75a3e99a72392e3934c768de